tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,559 +21,548 @@
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, -1.0]],
+                    expected=[
+                        [EXP_1, EXP_1, SINH_1],
+                        [EXP_1, EXP_1, SINH_1],
+                        [SINH_1, SINH_1, 1 / EXP_1],
+                    ],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inverse_differential_exp_data(self):
+            smoke_data = [
+                dict(
+                    tangent_vec=[
+                        [EXP_1, EXP_1, SINH_1],
+                        [EXP_1, EXP_1, SINH_1],
+                        [SINH_1, SINH_1, 1 / EXP_1],
+                    ],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, -1.0]],
+                    expected=[[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+    testing_data = TestDataSPDMatrices()
+
+    def test_belongs(self, n, mat, expected):
+        self.assertAllClose(SPDMatrices(n).belongs(gs.array(mat)), gs.array(expected))
+
+    def test_random_point(self, n, num_points):
+        space = SPDMatrices(n)
+        self.assertAllClose(gs.all(space.random_point(num_points)), gs.array(True))
+
+    def test_projection(self, n, mat, expected):
+        self.assertAllClose(
+            SPDMatrices(n).projection(gs.array(mat)), gs.array(expected)
         )
-        result = SPDMatrices(2).belongs(mats)
-        expected = gs.array([True, False, False])
-        self.assertAllClose(result, expected)
-
-    def test_random_point_and_belongs(self):
-        """Test of random_point and belongs methods."""
-        point = self.space.random_point()
-        result = self.space.belongs(point)
-        expected = True
-        self.assertAllClose(result, expected)
-
-    def test_random_point_and_belongs_vectorization(self):
-        """Test of random_point and belongs methods."""
-        points = self.space.random_point(4)
-        result = self.space.belongs(points)
-        expected = gs.array([True] * 4)
-        self.assertAllClose(result, expected)
-
-    def test_vector_from_symmetric_matrix_and_symmetric_matrix_from_vector(self):
-        """Test for matrix to vector and vector to matrix conversions."""
-        sym_mat_1 = gs.array([[1.0, 0.6, -3.0], [0.6, 7.0, 0.0], [-3.0, 0.0, 8.0]])
-        vector_1 = self.space.to_vector(sym_mat_1)
-        result_1 = self.space.from_vector(vector_1)
-        expected_1 = sym_mat_1
-
-        self.assertTrue(gs.allclose(result_1, expected_1))
-
-        vector_2 = gs.array([1.0, 2.0, 3.0, 4.0, 5.0, 6.0])
-        sym_mat_2 = self.space.from_vector(vector_2)
-        result_2 = self.space.to_vector(sym_mat_2)
-        expected_2 = vector_2
-
-        self.assertTrue(gs.allclose(result_2, expected_2))
-
-    def test_vector_and_symmetric_matrix_vectorization(self):
```

Comment:
This has been removed bcs it is tested in the symmetric matrices file, correct?

Reply:
yup